File: Leetcode/50.cpp
class Solution {
public:
    double myPow(double x, int n) {
        double ret = 1.0;
        int i = n;
        while (i) {
            if (i & 1) ret *= x;
            x *= x;
            i /= 2;
        }
        return n < 0 ? 1.0 / ret : ret;
    }
};

File: Leetcode/50.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        ret = 1.0
        i = int(n)
        while i != 0:
            if i & 1 == 1:
                ret *= x
            x *= x
            i = int(i / 2)
        if n < 0:
            return 1.0 / ret
        else:
            return ret
